File: tests/fixtures/generic/vulnerable/taint_sqli.java
// @audit-fixture
// @rule: taint_sqli
// @expected: detected
import javax.servlet.http.HttpServletRequest;
import java.sql.*;

public class UserController {
    public void findUser(HttpServletRequest request) throws Exception {
        String name = request.getParameter("name");
        Statement stmt = conn.createStatement();
        stmt.executeQuery("SELECT * FROM users WHERE name = '" + name + "'");
    }
}

File: tests/fixtures/generic/vulnerable/taint_sqli.js
// @audit-fixture
// @rule: taint_sqli
// @expected: detected
const express = require("express");
const app = express();
app.get("/users", (req, res) => {
    const name = req.query.name;
    db.query("SELECT * FROM users WHERE name = '" + name + "'");
});
